packages/flutter/lib/src/painting/debug_overflow_indicator.dart

```diff
@@ -0,0 +1,392 @@
+        0.0,
+        _overflowContainerRect.height * (1.0 - _kMarkerSize),
+        _overflowContainerRect.width,
+        _overflowContainerRect.height * _kMarkerSize,
+      );
+      regions.add(new _OverflowRegionData(
+        rect: markerRect,
+        label: 'BOTTOM OVERFLOWED BY $pixels PIXELS',
+        labelOffset: markerRect.bottomCenter -
+            const Offset(0.0, _debugMarkerFontSize + _debugMarkerLabelPadding),
+        rotation: 0.0,
+      ));
+    }
+    return regions;
+  }
+
+  void _reportOverflow() {
+    final List<String> overflows = <String>[];
+    if (overflow.left > 0.0)
+      overflows.add(_formatPixels(overflow.left) + ' pixels on the left');
+    if (overflow.top > 0.0)
+      overflows.add(_formatPixels(overflow.top) + ' pixels on the top');
+    if (overflow.bottom > 0.0)
+      overflows.add(_formatPixels(overflow.bottom) + ' pixels on the bottom');
+    if (overflow.right > 0.0)
+      overflows.add(_formatPixels(overflow.right) + ' pixels on the right');
+    String overflowText = '';
+    assert(overflows.isNotEmpty,
+        "Somehow $_containerName didn't actually overflow like it thought it did.");
+    switch (overflows.length) {
+      case 1:
+        overflowText = overflows.first;
+        break;
+      case 2:
+        overflowText = '${overflows.first} and ${overflows.last}';
+        break;
+      default:
+        overflows[overflows.length - 1] = 'and ${overflows[overflows.length - 1]}';
+        overflowText = overflows.join(', ');
+    }
+    FlutterError.reportError(
+      new FlutterErrorDetailsForRendering(
+        exception: 'A $_containerName overflowed by $overflowText.',
+        library: 'rendering library',
+        context: 'during layout',
+        renderObject: overflowRenderObject,
+        informationCollector: (StringBuffer information) {
+          information.writeln(overflowHints);
+          information.writeln('The specific $_containerName in question is:');
+          information.writeln('  ${overflowRenderObject.toStringShallow(joiner: '\n  ')}');
+          information.writeln('◢◤' * (FlutterError.wrapWidth ~/ 2));
+        },
+      ),
+    );
+  }
+
+  /// To be called when the overflow indicators should be painted.
+  ///
+  /// Typically only called if there is an overflow, and only from within a
+  /// debug build.
+  ///
+  /// See example code in [DebugOverflowIndicatorMixin] documentation.
+  void overflowPaintIndicator(PaintingContext context, Offset offset) {
+    assert(overflowRenderObject != null,
+        'Must set overflowRenderObject before calling overflowPaintIndicator',);
+    if (!isOverflowing) {
+      return;
+    }
+
+    final List<_OverflowRegionData> overflowRegions = _calculateOverflowRegions();
+    for (_OverflowRegionData region in overflowRegions) {
+      context.canvas.drawRect(region.rect.shift(offset), _debugMarkerPaint);
+
+      _debugMarkerLabel.text = new TextSpan(
+        // This is a no-op if the label hasn't changed.
```

Comment:
The region will change every time now if there's two or more regions.

Reply:
Yes, you're right.  Removed the comment.  Do you think it's worth it to have a four-element array with separate debugIndicatorLabels for each side in it that get updated?  Then it would go back to being a no-op if they haven't changed.